### ISSUE TITLE
node_pools_cgroup_mode var does not correctly handle the "all" map key

### ISSUE THREAD

**Issue by bboerst** [terraform-google-modules/terraform-google-kubernetes-engine#2196]
### TL;DR

For module users who want to pin the CGROUP version using the new `node_pools_cgroup_mode` feature released in #2001, specifying an "`all`" key does not correctly apply the value to all node-pools like expected.

Some examples...

This does not apply the change to any node pools.
```
node_pools_cgroup_mode = {
  all = "CGROUP_MODE_V2"
}
```

...but specifying the individual node pool names does work:
```
node_pools_cgroup_mode = {
  default-pool = "CGROUP_MODE_V2"
  ondemand = "CGROUP_MODE_V2"
  spot = "CGROUP_MODE_V2"
}
```

I believe that the logic for handling `all` is missing from here:
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/ef6c1ba6055f9f50c566fd4b72747fb1155a5569/modules/beta-private-cluster/cluster.tf#L880

### Expected behavior

The `all` key is nice for users who want to pin the CGROUP version for all node-pools without needing to specify each individual node-pool.

The default variable does include an `all` key, so I believe that the intention is to have this working:
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/ef6c1ba6055f9f50c566fd4b72747fb1155a5569/modules/beta-private-cluster/variables.tf#L243-L246

### Observed behavior

When specifying `all` for `node_pools_cgroup_mode`, no changes are made to the cluster configuration. When each individual node-pool is specified, then changes are made correctly.

### Terraform Configuration

```hcl
node_pools_cgroup_mode = {
  all = "CGROUP_MODE_V2"
}
```
```


### Terraform Version

```sh
`1.9.8.`
```

**Comment by DrFaust92** [comment]
Ill open a PR, I think I see the issue
